airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultCheckConnectionWorker.java

```diff
@@ -79,14 +91,25 @@ public StandardCheckConnectionOutput run(final StandardCheckConnectionInput inpu
 
         LOGGER.debug("Check connection job subprocess finished with exit code {}", exitCode);
         LOGGER.debug("Check connection job received output: {}", output);
-        return output;
+        return new ConnectorJobOutput().withOutputType(OutputType.CHECK_CONNECTION).withCheckConnection(output);
       } else {
         final String message = String.format("Error checking connection, status: %s, exit code: %d", status, exitCode);
-
         LOGGER.error(message);
-        return new StandardCheckConnectionOutput()
+
+        final Optional<AirbyteTraceMessage> traceMessage =
+            messagesByType.getOrDefault(Type.TRACE, new ArrayList<>()).stream()
+                .map(AirbyteMessage::getTrace)
+                .filter(trace -> trace.getType() == AirbyteTraceMessage.Type.ERROR)
+                .findFirst();
+
+        if (traceMessage.isPresent()) {
+          final FailureReason failureReason = FailureHelper.genericFailure(traceMessage.get(), null, null);
+          return new ConnectorJobOutput().withOutputType(OutputType.CHECK_CONNECTION).withFailureReason(failureReason);
+        }
```

Comment:
This logic seems to be repeated a few times - is there the opportunity to combine this traceMessage finding + the default error message in a utility that these workers all can share?

```java
final String defaultErrorMessage = String.format("Error checking connection, status: %s, exit code: %d", status, exitCode)
return WorkerUtils.buildReponseWithFailureReasons(OutputType type, messagesByType, string defaultErrorMessage)
```

Reply:
Good call! Done in [7eacb6e](https://github.com/airbytehq/airbyte/pull/14715/commits/7eacb6e2f77fdf6402a70a67645e11361366c167) 😄